cmake/list_insert_options.cmake

```diff
@@ -0,0 +1,55 @@
+#
+# Append key[=value] elements from one list to another list if they are not already in the target list.
```

Comment:
If I recall it correctly, the dependencies will be resolved first.
So changing some options might break ABI or even API.